src/studio/src/designer/backend/Models/AltinnStudioSettings.cs

```diff
@@ -8,9 +9,15 @@ namespace Altinn.Studio.Designer.Models
     public class AltinnStudioSettings
     {
         /// <summary>
-        /// The type of Altinn repository.
+        /// The type of Altinn repository ie. if this an app or datamodels repository.
         /// </summary>
         [JsonPropertyName("repoType")]
-        public string RepoType { get; set; }
+        public AltinnRepositoryType RepoType { get; set; }
+
+        /// <summary>
+        /// What kind of modelling schema that is used ie. Xsd or Json Schema.
+        /// </summary>
+        [JsonPropertyName("datamodelling.preference")]
```

Comment:
Is it necessary to have a period in the name? I was thinking that  `datamodellingPreference` is fine.

One thing to note about the designer backend is that it is using NewtonSoft to serialize and deserialize models.
You can see the statement `services.AddControllers().AddNewtonsoftJson();` in MvcConfiguration.cs.

My point being that any System.Text.Json attributes are ignored as far as I know.

Reply:
@altinnadmin - care to comment on the naming convension?

@SandGrainOne - thanks, you are right it's being ignored. I'll fix.

Reply:
The intention with the dot is to introduce a namespace for related settings. This model mirrors the .altinnstudio\settings.json file, which probably should be reflected a bit better in the controller. Perhaps change the name to StudioSettingsController ?